spacepy/pybats/pwom.py

```diff
@@ -5,142 +5,110 @@
-        self.attrs['nLat']=nLat
-        self.attrs['nLon']=nLon
+        self.attrs['nLat'] = nLat
+        self.attrs['nLon'] = nLon
 
         # Create arrays.
         for k, u in varlist:
-            self[k]=dmarray(np.zeros( (nLat*nLon) ), attrs={'units':u})
+            self[k] = dmarray(np.zeros((nLat*nLon)), attrs={'units': u})
 
         for i in range(nLat*nLon):
-            #for j in range(nLon):
-            parts=f.readline().split()
+            parts = f.readline().split()
             for k, (key, u) in enumerate(varlist):
-                self[key][i]=float(parts[k])
+                self[key][i] = float(parts[k])
 
-        # Lat, CoLat, and Lon:
-        #self['lon'] = dmarray(np.pi/2 - np.arctan(self['y']/self['x']),
-        #                      attrs={'units','deg'})*180.0/np.pi
-        #self['lon'][self['x']==0.0] = 0.0
-        xy=np.sqrt(self['x']**2+self['y']**2)+0.0000001
-        self['lon']=np.arcsin(self['y']/xy)
-        self['lat']  =dmarray(np.arcsin(self['z']), {'units':'deg'})*180.0/np.pi
-        self['colat']=90.0 - self['lat']
+        xy = np.sqrt(self['x']**2+self['y']**2)+0.0000001
+        self['lon'] = np.arcsin(self['y']/xy)
+        self['lat'] = dmarray(np.arcsin(self['z']),
+                              {'units': 'deg'})*180.0/np.pi
+        self['colat'] = 90.0 - self['lat']
 
         f.close()
 
-class Line(PbData):
+
+class Line(IdlFile):
     '''
     Class for loading a single field line output file.
-    At instantiation time, user may wish to set the start date and time of 
+    At instantiation time, user may wish to set the start date and time of
     the simulation using the starttime kwarg.  If not given, start time
     will default to Jan. 1st, 2000, 00:00UT.
+
+    Parameters
+    ----------
+    filename : str
+        Name of line file to open
+    starttime : datetime.datetime
+        Simulation time corresponding to line file.
     '''
     def __init__(self, filename, starttime=None, *args, **kwargs):
-        super(Line, self).__init__(*args, **kwargs) # Init as PbData.
-        self.attrs['file']=filename
+        # Initialize as an IdlFile. Filename handled by parent reader and
+        # stashed as self.attrs['filename']
+        super(Line, self).__init__(filename, header=None, *args, **kwargs)
+
         if not starttime:
-            starttime=dt.datetime(2000,1,1,0,0,0)
-        self._read(starttime)
+            starttime = dt.datetime(2000, 1, 1, 0, 0, 0)
 
-    def __repr__(self):
-        return 'PWOM single field line output file %s' % (self.attrs['file'])
+        # Parse PWOM's units from variable names.
+        for v in self.keys():
+            if v == 'grid':
+                continue
+            matched = re.match('(\w+)\[(.*)\]', v)
+            if matched:
+                self[matched.groups()[0]] = self.pop(v)
+                self[matched.groups()[0]].attrs['units'] = matched.groups()[1]
 
-    def _read(self, starttime):
```

Comment:
I presume this all went away from PWOM because the generic ASCII reading went into IdlFile?